tools/fdleak_example.txt

```diff
@@ -0,0 +1,140 @@
+         PNAME-PID            LIMIT     MAX_SURVIVAL(s) MAX_FD   THREADs NAME-TID list
+         test_fd_leak-85485   unlimited 35              175      2       fdleak_2-85486 / fdleak_3-85491
+         Backtrace:
+         open+0xd4 [libpthread-2.31.so]
+         [unknown]
+total 0
+dr-x------ 2 vachel.yang vachel.yang  0 oct 16 17:49 .
+dr-xr-xr-x 9 vachel.yang vachel.yang  0 oct 16 17:48 ..
+lrwx------ 1 vachel.yang vachel.yang 64 oct 16 17:49 0 -> /dev/pts/1
+lrwx------ 1 vachel.yang vachel.yang 64 oct 16 17:49 1 -> /dev/pts/1
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 10 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 100 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 101 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 102 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 103 -> /usr/share/code/v8_context_snapshot.bin
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 104 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 105 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 106 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 107 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 108 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 109 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 11 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 110 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 111 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 112 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 113 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 114 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 115 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 116 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 117 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 118 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 119 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 12 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 120 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 121 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 122 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 123 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 124 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 125 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 126 -> /dev/null
+lr-x------ 1 vachel.yang vachel.yang 64 oct 16 17:49 127 -> /dev/null
+
+---->
+^ fdleak thread open too much `/dev/null`
+----
+Currently i don't know how to deal with the case that recycling program resources by system `do_exit` -> `exit_files` . 
```

Comment:
Yes! This is a real problem...can we kprobe filp_close() to handle sources from __close_fd and also exit_files?